src/jit/hwintrinsiclistarm64.h

```diff
@@ -91,6 +92,8 @@ HARDWARE_INTRINSIC(ArmBase,         LeadingZeroCount,                          -
 //  Base 64-bit only Intrinsics
 HARDWARE_INTRINSIC(ArmBase_Arm64,   LeadingSignCount,                          -1,               0,           1,     {INS_invalid,           INS_invalid,        INS_invalid,        INS_invalid,        INS_cls,            INS_invalid,        INS_cls,            INS_invalid,        INS_invalid,        INS_invalid},           HW_Category_Scalar,                 HW_Flag_NoContainment)
 HARDWARE_INTRINSIC(ArmBase_Arm64,   LeadingZeroCount,                          -1,               0,           1,     {INS_invalid,           INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_clz,            INS_clz,            INS_invalid,        INS_invalid},           HW_Category_Scalar,                 HW_Flag_NoContainment)
+HARDWARE_INTRINSIC(ArmBase_Arm64,   ReverseElementBits,                        -1,               0,           1,     {INS_invalid,           INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_rbit,           INS_rbit,           INS_invalid,        INS_invalid},           HW_Category_Scalar,                 HW_Flag_NoContainment)
+HARDWARE_INTRINSIC(AdvSimd_Arm64,   ReverseElementBits,                        -1,               16,          1,     {INS_rbit,              INS_rbit,           INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid},           HW_Category_SimpleSIMD,             HW_Flag_NoContainment)
```

Comment:
1. Can you please move the line corresponding to `AdvSimd.Arm64.ReverseElementBits` to *AdvSimd 64-bit only Intrinsics* group above
2. Since AdvSimd version of `rbit` supports both uint8x8_t and uint8x16_t shouldn't the `SIMD size` be -1 and marked as `HW_Flag_UnfixedSIMDSize`? @tannergooding @CarolEidt can you please advise

Reply:
On 2, that seems to be correct; otherwise presumably this will always generate the 16-byte version.

Reply:
Yes, that should be the case. I'm actually surprised this isn't failing due to 2 enum entries with the same name...

Reply:
Ah, nevermind, the class is different, it's just in the wrong place

Reply:
hmm, 
```
G_M18293_IG04:
        FD4017A0          ldr     d0, [fp,#40]
        6E605800          rbit    v0.16b, v0.16b

```
Right it does, I didn't look at the codegen for this one (in terms of correctness the wider vector didn't matter in this case).

I had initially understood this as being the maximum vector length. That said why is `HW_Flag_UnfixedSIMDSize` not the default? I assume this is an optimization, but at least on Arm/AArch64 the vast majority of the intrinsics would require this.

Wouldn't allowing to overload the HWintrinsics list based on SIMD size as well solve the performance issue then? And make it easier to specify intrinsics that don't support a particular type in a vector length but do in the other? such as `Vector64<long>` not being supported but `Vector128<long>` is in some cases?

Also I didn't quite understand why Scalar intrinsics are excluded from the table driven approach requiring them to be handled individually like above.

Reply:
The infrastructure is shared with and largely ported from x86, which is why some "defaults" don't necessarily make sense when viewed across both platforms.

For some of them, we could likely simplify things. For example, we could remove the `UnfixedSIMDSize` flag in favor of just having the size field be `-1`. -- We are also likely missing an assert that validates the specified SIMD size and the argument size is consistent...

> Wouldn't allowing to overload the HWintrinsics list based on SIMD size as well solve the performance issue then

I don't think we have a performance issue here with the current table (aside from matching method names to IDs, which would likely be worsened due to more entries in the table and additional logic needed to handle that).

> Also I didn't quite understand why Scalar intrinsics are excluded from the table driven approach requiring them to be handled individually like above.

x86 had very few scalar intrinsics and the first iteration of the table/logic couldn't handle both. The current iteration should be able to, but we just haven't prioritized that work yet (I believe we have an issue tracking it).

Reply:
> I believe we have an issue tracking it.

Yes, correct - I opened https://github.com/dotnet/coreclr/issues/27752 to track the work getting importation of intrinsics to be table-driven. Assigned this to myself. Similar work needs to be done for code-generation as well .